clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -5407,6 +5407,9 @@ def note_dependent_function_template_spec_discard_reason : Note<
   "candidate ignored: %select{not a function template|"
   "not a member of the enclosing %select{class template|"
   "namespace; did you mean to explicitly qualify the specialization?}1}0">;
+def warn_diag_specialization : Warning<
```

Comment:
nit: `warn_diag_...` is a bit of a weird name for a diagnostic. Can we call this just `warn_invalid_specialization` or sth like that?